geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaConsumerFeatureSource.scala

```diff
@@ -65,7 +67,20 @@ class KafkaConsumerFeatureSource(entry: ContentEntry,
     }
   }
 
-  val features = Maps.newConcurrentMap[String, FeatureHolder]()
+  val cb = CacheBuilder.newBuilder()
+
+  if (expiry) {
+    cb.expireAfterWrite(expirationPeriod, TimeUnit.MILLISECONDS)
+      .removalListener(
+        new RemovalListener[String, FeatureHolder] {
+          def onRemoval(removal: RemovalNotification[String, FeatureHolder]) = {
+            removeFeature(new Delete(removal.getKey))
```

Comment:
Maybe add a test that doesn't call cleanup() on the cache and see if it works or not.